aptos-move/e2e-move-tests/src/tests/token_objects.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, tests::common, MoveHarness};
+use aptos_types::{
+    account_address::{self, AccountAddress},
+    event::EventHandle,
+};
+use move_core_types::{identifier::Identifier, language_storage::StructTag};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+struct Token {
+    collection: String,
+    creator: AccountAddress,
+    description: String,
+    mutability_config: MutabilityConfig,
+    name: String,
+    creation_name: Option<String>,
+    uri: String,
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+struct MutabilityConfig {
+    description: bool,
+    name: bool,
+    uri: bool,
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+struct Object {
+    guid_creation_num: u64,
+    owner: AccountAddress,
+    allow_ungated_transfer: bool,
```

Comment:
I wonder if there should be a built-in revocation mechanism for capabilities like Transfer. In this case we need to store here more. But it is doable (we have implemented those for Capabilities back at Diem).

Reply:
We probably cannot revoke, but we can at least disable. The beauty of the resource group approach is that we can defer decisions to a later date and only pay a negligible performance cost.

From my vantage, the intent of the TransferRef is for the creator to dictate the behavior. Of course they could share that out, I might recommend that they wrap it in something else that is revocable. I say that because I don't see revocation of this very common -- I imagine it is more used for assisted management of objects and maybe it works into our larger capabilities model.

Reply:
Definitely we can revoke a TransferRef. See other comment.

Reply:
I agree with david that it should be up to the user to decide- revocation of transfer feels like a very use specific item, and is too specific for a primitive here

On that note, I think the concept of an ungated/gated transfer is also too use-case specific, and feels like generally upper levels will come up with their own versions of it